kotlin-runtime/ftl-runtime/src/main/kotlin/xyz/block/ftl/schemaextractor/ExtractSchemaRule.kt

```diff
@@ -214,44 +214,55 @@ class SchemaExtractor(
       }
 
       // Validate parameters
-      require(verb.valueParameters.size == 2) { "$verbSourcePos Verbs must have exactly two arguments, ${verb.name} did not" }
+      require(verb.valueParameters.size >= 1) { "$verbSourcePos Verbs must have at least one argument, ${verb.name} did not" }
+      require(verb.valueParameters.size <= 2) { "$verbSourcePos Verbs must have at most two arguments, ${verb.name} did not" }
       val ctxParam = verb.valueParameters.first()
-      val reqParam = verb.valueParameters.last()
+
       require(ctxParam.typeReference?.resolveType()?.fqNameOrNull()?.asString() == Context::class.qualifiedName) {
         "${verb.valueParameters.first().getLineAndColumn()} First argument of verb must be Context"
       }
 
-      require(reqParam.typeReference?.resolveType()
-        ?.let { it.toClassDescriptor().isData || it.isEmptyBuiltin() }
-        ?: false
-      ) {
-        "${verb.valueParameters.last().getLineAndColumn()} Second argument of ${verb.name} must be a data class or " +
-          "builtin.Empty"
+      if (verb.valueParameters.size == 2) {
+        val reqParam = verb.valueParameters.last()
+        require(reqParam.typeReference?.resolveType()
+          ?.let { it.toClassDescriptor().isData || it.isEmptyBuiltin() || it.isUnit() }
```

Comment:
do we want to allow kotlin.Unit as a request type? i thought we were going to infer this from omission with the new change

Reply:
Ahh good catch! I had this in before I added the size check above and forgot to remove! Thanks for pointing this one out!